filters/auth/oidc.go

```diff
@@ -198,6 +247,14 @@ func (s *tokenOidcSpec) CreateFilter(args []interface{}) (filters.Filter, error)
 		encrypter:          encrypter,
 		compressor:         newDeflatePoolCompressor(flate.BestCompression),
 		subdomainsToRemove: subdomainsToRemove,
+		distributedClaimsClient: snet.NewClient(snet.Options{
```

Comment:
This will leak goroutines, we need to use static client or cache clients similar to https://github.com/zalando/skipper/pull/1967

Reply:
@AlexanderYastrebov If so that would mean GC is not able to purge the filter instance once released - then other resources would be leaking as well

Reply:
This goroutine https://github.com/zalando/skipper/blob/2c79795de8f376d34e5b4fecdcc58718cd90aeb6/net/httpclient.go#L282-L291 will never finish because we do not call `Close()` on the client so the number of goroutines will steadily grow over time due to route updates.
There is #202 to support filter resource cleanup but it is not fixed yet. The workaround for now is to cache clients per url like it is done in tokeninfo or webhook filters.

Reply:
Thanks for the context, that is indeed a leak.

So far I can't see how the caching of the client per URL is applicable here, since every URL will most likely be a user-unique call, plus the credentials might clash when it is global...
Storing the snet.client as a global variable sounds like the only option?

Reply:
it's cached by host

Reply:
Sorry guess I'm not getting it :) @szuecs  but that does not hold true at least for the tokeninfo where the URL including path is the key:

```go
	if ac, ok = tokeninfoAuthClient[s.options.URL]; !ok {
		ac, err = newAuthClient(s.options.URL, ...
```
https://github.com/zalando/skipper/blob/master/filters/auth/tokeninfo.go#L183-L184

But if you asking for this pattern we can do that here

Reply:
yes, because there is only one tokeninfo URL per skipper.
For OIDC I would do one per host, so parse URL string to URL and use the url.Host as a key to cache the client.

Reply:
Ha good point, gotcha!
Implemented, please review